Quadriga/src/main/java/edu/asu/spring/quadriga/dao/impl/workbench/RetrieveProjectDAO.java

```
@@ -361,5 +361,30 @@ public ProjectDTO getDTO(String id) {
     }
 	
 	
+    /**
+     * 
+     * This method fetches all the  projects with the given accessibility
+     * 
+     * Uses Hibernate to get {@link ProjectDTO} of a {@link IProject} ID. 
+     */
+    @SuppressWarnings("unchecked")
+    @Override
+    public List<ProjectDTO> getAllProjectsDTOByAccessibility(String accessibility) throws QuadrigaStorageException 
+    {
+    	List<ProjectDTO> projectDTOList = null;
+    	try
+    	{
+    		Query query = sessionFactory.getCurrentSession().createQuery("from ProjectDTO project where project.accessibility = :accessibility");
+    		query.setParameter("accessibility", accessibility);
+
+    		projectDTOList =  query.list();
+    		return projectDTOList;
+    	}
+    	catch(Exception e)

```

don't catch all exception